src/TestSuite/Migrator.php

```diff
@@ -0,0 +1,253 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Migrations\TestSuite;
+
+use Cake\Console\ConsoleIo;
+use Cake\Datasource\ConnectionManager;
+use Cake\TestSuite\Schema\SchemaCleaner;
+use Cake\TestSuite\TestConnectionManager;
+use Migrations\Migrations;
+
+class Migrator
+{
+    /**
+     * @var ConfigReader
+     */
+    protected $configReader;
+
+    /**
+     * @var ConsoleIo
+     */
+    protected $io;
+
+    /**
+     * Migrator constructor.
+     * @param bool $verbose
+     * @param null $configReader
+     */
+    final public function __construct(bool $verbose, ?ConfigReader $configReader = null)
```

Comment:
Why is this function final?

Reply:
I was originally thinking to have the migrator extending the Cake's SchemaManager. This is now done, so there is no construct needed anymore.